OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -821,7 +821,9 @@ public void ResolveOrder(Actor self, Order order)
 			else if (order.OrderString == "ReturnToBase" && rearmableInfo != null && rearmableInfo.RearmActors.Any())
 			{
 				UnReserve();
-				self.CancelActivity();
+				if (!order.Queued)
+					self.CancelActivity();
+
 				if (!Info.CanHover)
 					self.QueueActivity(new ReturnToBase(self, Info.AbortOnResupply, null, false));
```

Comment:
Actually, this can be done in a better way:
Actor.QueueActivity has a `(bool queued, Activity nextActivity)` overload that cancels current activity if RTB isn't queued, so
`self.QueueActivity(order.Queued, new ReturnToBase(self, Info.AbortOnResupply, null, false));`
will give the same result with less lines.

Reply:
Done.